File: app/src/main/java/calendario/kevshupp/diariokevinali/CalendarEvent.java
package calendario.kevshupp.diariokevinali;

public class CalendarEvent {
    private String eventId;
    private String title;
    private String description;
    private long date; 
    private String authorId;
    private String authorName;
    private String partnerId;
    private String recurrence; // "NONE", "WEEKLY", "YEARLY"

    public CalendarEvent() {
        this.recurrence = "NONE";
    }

    public CalendarEvent(String eventId, String title, String description, long date, String authorId, String partnerId) {
        this.eventId = eventId;
        this.title = title;
        this.description = description;
        this.date = date;
        this.authorId = authorId;
        this.partnerId = partnerId;
        this.recurrence = "NONE";
    }

    public String getEventId() { return eventId; }
    public void setEventId(String eventId) { this.eventId = eventId; }

    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public long getDate() { return date; }
    public void setDate(long date) { this.date = date; }

    public String getAuthorId() { return authorId; }
    public void setAuthorId(String authorId) { this.authorId = authorId; }

    public String getAuthorName() { return authorName; }
    public void setAuthorName(String authorName) { this.authorName = authorName; }

    public String getPartnerId() { return partnerId; }
    public void setPartnerId(String partnerId) { this.partnerId = partnerId; }

    public String getRecurrence() { return recurrence; }
    public void setRecurrence(String recurrence) { this.recurrence = recurrence; }
}

File: app/src/main/java/calendario/kevshupp/diariokevinali/CalendarEvent.kt
package calendario.kevshupp.diariokevinali

import androidx.compose.runtime.Stable

/**
 * Modelo para eventos del calendario con soporte para recurrencia.
 */
@Stable
data class CalendarEvent(
    var eventId: String = "",
    var title: String = "",
    var description: String = "",
    var date: Long = 0,
    var authorId: String = "",
    var authorName: String = "",
    var partnerId: String = "",
    var recurrence: String = "NONE" // "NONE", "WEEKLY", "YEARLY"
) {
    // Constructor secundario para compatibilidad con el código Java existente (6 parámetros)
    constructor(
        eventId: String,
        title: String,
        description: String,
        date: Long,
        authorId: String,
        partnerId: String
    ) : this(eventId, title, description, date, authorId, "", partnerId, "NONE")
}
